src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBwaAligner.java

```diff
@@ -0,0 +1,100 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFlag;
+import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.*;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+import scala.Tuple2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Loads Bwa index and aligns reads. This is not a general Bwa aligner because it only retrieves primary alignments.
+ * If preserveAlignments is set, existing alignment data in each read is not overwritten, and any new alignments
+ * are added to the SA tag.
+ */
+public final class PSBwaAligner {
+
+    private final BwaMemIndex bwaIndex;
+    private final int minSeedLength, numThreads, maxAltHits, scoreThreshold;
+    private final boolean pairedAlignment;
+
+    public PSBwaAligner(final PSBwaArgumentCollection bwaArgs, final boolean pairedAlignment) {
```

Comment:
Might as well just save the bwaArgs collection, rather than copying each value into this class.